syntax = "proto3";
package collection_config;




import "google/protobuf/struct.proto";

/*
ListCollectionConfigJobs请求
*/
message ListCollectionConfigJobsRequest {
    /*
    配置ID
    */
    string confId = 1 ;
}

/*
ListCollectionConfigJobs返回
*/
message ListCollectionConfigJobsResponse {
message List {
message Host {
    /*
    主机
    */
    string instanceId = 1 ;
    /*
    IP
    */
    string ip = 2 ;
    /*
    Agent唯一标识
    */
    string uuid = 3 ;
}
        
message Script {
    /*
    脚本Id
    */
    string id = 1 ;
    /*
    脚本版本
    */
    string version = 2 ;
    /*
    脚本类型
    */
    string type = 3 ;
    /*
    脚本内容
    */
    string content = 4 ;
    /*
    脚本名称
    */
    string name = 5 ;
}
        
message Env {
    /*
    键
    */
    string key = 1 ;
    /*
    值
    */
    google.protobuf.Value value = 2 ;
}
        
message Kwargs {
    /*
    键
    */
    string key = 1 ;
    /*
    值
    */
    google.protobuf.Value value = 2 ;
}
        
    
    /*
    执行时间间隔
    */
    int32 interval = 1 ;
    /*
    超时间隔
    */
    int32 timeout = 2 ;
    /*
    执行时间段
    */
    string timeRange = 3 ;
    /*
    主机
    */
    Host host = 4 ;
    /*
    指标表名
    */
    string metricTable = 5 ;
    /*
    插件ID
    */
    string pluginId = 6 ;
    /*
    采集作业名称
    */
    string name = 7 ;
    /*
    脚本
    */
    Script script = 8 ;
    /*
    数据ID
    */
    int32 dataId = 9 ;
    /*
    环境变量
    */
    repeated Env env = 10 ;
    /*
    参数
    */
    repeated Kwargs kwargs = 11 ;
    /*
    执行方法
    */
    string fun = 12 ;
    /*
    采集类型ID
    */
    string clazzId = 13 ;
    /*
    采集类型名称
    */
    string clazzName = 14 ;
    /*
    采集任务ID
    */
    string jobConfigId = 15 ;
    /*
    数据要求字段
    */
    repeated string requiredFields = 16 ;
    /*
    缓存过期次数（时间）
    */
    int32 cacheTtl = 17 ;
    /*
    是否忽略不符合要求数据
    */
    bool ignoreInvalid = 18 ;
    /*
    标签
    */
    repeated string labels = 19 ;
    /*
    是否禁用
    */
    bool disabled = 20 ;
    /*
    ORG
    */
    int32 org = 21 ;
    /*
    采集作业ID
    */
    string id = 22 ;
    /*
    创建者
    */
    string creator = 23 ;
    /*
    修改者
    */
    string modifier = 24 ;
    /*
    创建时间
    */
    int32 ctime = 25 ;
    /*
    修改时间
    */
    int32 mtime = 26 ;
    /*
    关联模型ID
    */
    string objectId = 27 ;
    /*
    关联模型实例ID
    */
    string instanceId = 28 ;
}
        
    
    /*
    页数
    */
    int32 page = 1 ;
    /*
    页大小
    */
    int32 page_size = 2 ;
    /*
    总数
    */
    int32 total = 3 ;
    /*
    数据列表
    */
    repeated List list = 4 ;
}

/*
ListCollectionConfigJobsApi返回
*/
message ListCollectionConfigJobsResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    collection_config.ListCollectionConfigJobsResponse data = 4 ;
}

